conjure-go-client/httpclient/client.go

```diff
@@ -97,14 +97,15 @@ func (c *clientImpl) Do(ctx context.Context, params ...RequestParam) (*http.Resp
 	failedURIs := map[string]struct{}{}
 	for i := 0; i < c.maxRetries; i++ {
 		resp, err = c.doOnce(ctx, nextURI, params...)
-		if resp == nil {
-			// If we get a nil response, we can assume there is a problem with host and can move on to the next.
-			nextURI, offset = nextURIOrBackoff(nextURI, uris, offset, failedURIs, retrier)
-		} else if shouldThrottle, _ := internal.IsThrottleResponse(resp); shouldThrottle {
+		if retryOther, _ := internal.IsThrottleResponse(resp, err); retryOther {
```

Comment:
These if blocks are  are reordered now.. I guess there is no functionality change though? Can you confirm?

Reply:
no, this does matter now because the `resp == nil` case is true for both this statement and the following one. The test I added will fail if the order changes